### PR TITLE
feat: handle draft seats and entitlements in ecommerce data loader

### PR DESCRIPTION
**Pull request by zawan-ila**
## [PROD-3509](https://2u-internal.atlassian.net/browse/PROD-3509)

In this PR we modify the Ecommerce Data Loader to update `draft` versions of seats and entitlements together with the `non-draft`ones.

## Testing

1. Update some seat or entitlement information in Ecommerce.
2. Run RCM
3. Verify that both the official and draft versions of the objects(seats/entitlements) have been updated.

### CODE REVIEW

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 1]
```diff
@@ -601,13 +605,24 @@ def update_seat(self, course_run, product_body):
...
-        _, created = course_run.seats.update_or_create(
+        seat, created = course_run.seats.update_or_create(
             type=seat_type,
             credit_provider=credit_provider,
             currency=currency,
             defaults=defaults
         )
 
+        if course_run.draft_version:
+            draft_seat, created = course_run.draft_version.seats.update_or_create(
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/data_loaders/api.py]: nit: created will override the previous created. If created is being used in multiple places, ensure that this does not cause any unintended consequences.

File: course_discovery/apps/course_metadata/data_loaders/tests/test_api.py [hunk 2]
```diff
@@ -640,6 +640,7 @@ def mock_products_api(self, alt_course=None, alt_currency=None, alt_mode=None, h
                           valid_stockrecord=True, product_class=None):
         """ Return a new Course Entitlement and Enrollment Code to be added by ingest """
         course = CourseFactory(type=CourseType.objects.get(slug=CourseType.VERIFIED_AUDIT), partner=self.partner)
+        ensure_draft_world(Course.objects.get(pk=course.pk))
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/data_loaders/tests/test_api.py]: nit: Why is Course fetched again?

**Review by Ali-D-Akbar — APPROVED** [review]

File: course_discovery/apps/course_metadata/data_loaders/tests/test_api.py (hunk 2 above)
Comment (zawan-ila) [reply course_discovery/apps/course_metadata/data_loaders/tests/test_api.py]: fixed

File: course_discovery/apps/course_metadata/data_loaders/api.py (hunk 1 above)
Comment (zawan-ila) [reply course_discovery/apps/course_metadata/data_loaders/api.py]: fixed

File: course_discovery/apps/course_metadata/data_loaders/tests/test_api.py [hunk 3]
```diff
@@ -787,6 +788,8 @@ def assert_seats_loaded(self, body, mock_products):
         products = [p for p in body['products'] if p['structure'] == 'child']
         # Verify that the old seat is removed
         assert course_run.seats.count() == len(products)
+        assert course_run.draft_version.seats.count() == len(products)
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/data_loaders/tests/test_api.py:791]: nit: alongside length, add assertions on values (sku & type importantly)

File: course_discovery/apps/course_metadata/data_loaders/tests/test_api.py [hunk 4]
```diff
@@ -907,7 +926,11 @@ def test_ingest_deletes(self, mock_logger):
         msg = 'Deleting entitlement for course {course_title} with sku {sku} for partner {partner}'.format(
             course_title=entitlement.course.title, sku=entitlement.sku, partner=entitlement.partner
         )
+        msg_draft = 'Deleting entitlement for course {course_title} with sku {sku} for partner {partner}'.format(
+            course_title=entitlement_draft.course.title, sku=entitlement_draft.sku, partner=entitlement_draft.partner
+        )
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/data_loaders/tests/test_api.py]: How much effort is involved in actually updating the message to include the Draft? The current message is same as non_draft, only that draft values are being added in the string.

**Review by DawoudSheraz — APPROVED** [review]
Tested locally for 2 separate courses. Things are looking in place, and the timestamp is also updating. 
Left a few nits/questions to address before merge.

File: course_discovery/apps/course_metadata/data_loaders/tests/test_api.py (hunk 3 above)
Comment (zawan-ila) [reply course_discovery/apps/course_metadata/data_loaders/tests/test_api.py:791]: This is already being done a few lines later.

File: course_discovery/apps/course_metadata/data_loaders/tests/test_api.py (hunk 4 above)
Comment (zawan-ila) [reply course_discovery/apps/course_metadata/data_loaders/tests/test_api.py]: Updated message to differentiate between draft and non-draft version

**Pull request merged**